general/macros.hpc

```diff
@@ -201,11 +201,11 @@ EOF\
 %hpc_module_dep_base    %{hpc_module_dep_path}/%{hpc_module_pname}%{?hpc_ext}/
 
 # For extra python flavors
-%hpc_module_dep_extra() \
+%hpc_module_dep_extra \
 %{lua: \
-local arg = rpm.expand("%1") \
-for extra in rpm.expand("%{" .. arg .. "_provides}"):gmatch("%S+") do \
-  print(rpm.expand("%hpc_module_dep_base"):gsub(arg, extra):sub(1,-2) .. "\\n") \
+local pyflavor = rpm.expand("%python_flavor") \
+for extra in rpm.expand("%{?" .. pyflavor .. "_provides}"):gmatch("%S+") do \
+  print(rpm.expand("%hpc_module_dep_base"):gsub(pyflavor, extra):sub(1,-2) .. "\\n") \
 end \
```

Comment:
No, sorry, now it becomes even more 'pythony'. 
if you decide to go this route, please move it to the other python specific macros and  rename it so that this becomes obvious.